### ISSUE TITLE
Best way to work with TreatWarningsAsErrors?

### ISSUE THREAD

**Issue by yhvicey** [DotNetAnalyzers/StyleCopAnalyzers#2906]
When using StyleCop with `TreatWarningsAsErrors`, it will always treat StyleCop's warning as error. Is there anyway to make all stylecop's warning not been treated as error?

Maybe `WarningsNotAsErrors` property would help resolve this - For example, StyleCop provides a full list of its warnings for users to append, or provide a switch to control if it should append the list to the property.

E.g.

```xml
<!-- In StyleCop nuget package -->
<PropertyGroup>
  <StyleCopWarningList>SA1000,SA1001,...</StyleCopWarningList>
</PropertyGroup>
<!-- In user code -->
<PropertyGroup>
  <WarningsNotAsErrors>$(WarningsNotAsErrors);$(StyleCopWarningList)</StyleCopWarningList>
</PropertyGroup>
```

**Comment by vweijsters** [comment]
You can change the severity of the StyleCop.Analyzers diagnostics to info using a custom rule set.

There is really not a lot that we can do about it, as the severity stuff is handled by Roslyn / Visual Studio.

**Comment by philipborg** [comment]
@yhvicey 
Yes it is possible to not raise specific warnings. I have done it at work, will check the tag you need tomorrow when I get back there. 👍 I personally configured it not to raise #warning pragma and obsolete.
